src/User/ProfilePage.test.jsx

```diff
@@ -16,32 +16,134 @@ jest.mock("react-router-dom", () => {
+  };
+
+  it("renders current user's profile information with manager", async () => {
+    expect.assertions(7);
+
+    const history = createMemoryHistory();
+    history.push("/users/11");
+
+    wrapper = mount(
+      <Router history={history}>
+        <ProfilePage
+          user={mockDataWithManager}
+          isCurrentUser={true}
+          managerName={mockManagerName}
+        />
+      </Router>
+    );
+
+    wrapper.update();
+    expect(wrapper.find(".editLink").get(0).props.to).toBe(
+      `/users/${mockDataWithManager.id}/edit`
+    );
+    expect(findElement(".name")).toBe(
+      `Hello, ${mockDataWithManager.firstName} ${mockDataWithManager.lastName}!`
+    );
+    expect(findElement(".username")).toBe(
+      `Username: ${mockDataWithManager.username}`
+    );
+    expect(findElement(".email")).toBe(`Email: ${mockDataWithManager.email}`);
+    expect(findElement(".occupation")).toBe(
+      `Occupation: ${mockDataWithManager.occupation}`
+    );
+    expect(findElement(".managerLink")).toBe(mockManagerName);
+    expect(wrapper.find(".managerLink").get(0).props.to).toBe(
+      `/users/${mockDataWithManager.managerId}`
+    );
+  });
+
+  it("renders current user's profile information without manager", async () => {
+    expect.assertions(7);
 
     const history = createMemoryHistory();
     history.push("/users/11");
 
-   const wrapper = mount(
-        <Router history={history}>
-          <ProfilePage user={mockData} />
-        </Router>
-      );
-
-      wrapper.update();
-    expect(wrapper.find(".editLink").get(0).props.to).toBe(`/users/edit/${mockData.id}`);     
-    expect(wrapper.find(".name").get(0).props.children).toBe(`Hello, ${mockData.firstName} ${mockData.lastName}!`);
-    expect(wrapper.find(".username").get(0).props.children).toBe(`Username: ${mockData.username}`);
-    expect(wrapper.find(".email").get(0).props.children).toBe(`Email: ${mockData.email}`);
-    expect(wrapper.find(".occupation").get(0).props.children).toBe(`Occupation: ${mockData.occupation}`);
+    wrapper = mount(
+      <Router history={history}>
+        <ProfilePage
+          user={mockDataWithoutManager}
+          isCurrentUser={true}
+          managerName={mockNoManagerName}
+        />
+      </Router>
+    );
+
+    wrapper.update();
+    expect(wrapper.find(".editLink").get(0).props.to).toBe(
+      `/users/${mockDataWithoutManager.id}/edit`
+    );
+    expect(findElement(".name")).toBe(
+      `Hello, ${mockDataWithoutManager.firstName} ${mockDataWithoutManager.lastName}!`
+    );
+    expect(findElement(".username")).toBe(
+      `Username: ${mockDataWithoutManager.username}`
+    );
+    expect(findElement(".email")).toBe(
+      `Email: ${mockDataWithoutManager.email}`
+    );
+    expect(findElement(".occupation")).toBe(
+      `Occupation: ${mockDataWithoutManager.occupation}`
+    );
+    expect(findElement(".manager")).toBe(`Manager: ${mockNoManagerName}`);
+    expect(wrapper.find(".managerLink").length).toBe(0);
```

Comment:
If you use a snapshot test, you're probably safe just checking the that child contains `Ross Morey` for example rather than the more verbose `Hello, ...`
If you correctly use snapshots and look at what is snapshotted, you will be able to tell if Hello isn't present or suddenly gets removed during future work